probe-rs/src/architecture/arm/core/cortex_m.rs

```diff
@@ -131,6 +131,139 @@ impl MemoryMappedRegister for Cpacr {
     const NAME: &'static str = "CPACR";
 }
 
+bitfield! {
+    #[derive(Copy, Clone)]
+    /// IPSR - Interrupt Program Status Register (a variant of xPSR)
+    pub struct Ipsr(u32);
+    impl Debug;
+    pub exception, _: 8,0;
+}
+
+/// When returning from an exception, the processor uses the EXC_RETURN value to determine which stack and mode to return to.
```

Comment:
@Tiwalun These enums and their impl's are cortex-m specific, and in some cases (e.g. Armv6m) requires even more tweaking. 

I would like to create something more generic, that can be used in `probes_rs::debug` code to unwind exception/trap handlers for RISC-V and other Arm variants.

Should I do this in the core interface, or should I keep the code contained to the `probes_rs::debug` module?